Vungle/VungleAdapter/ALVungleMediationAdapter.m

```diff
@@ -518,25 +517,49 @@ - (void)loadVungleNativeAdForParameters:(id<MAAdapterResponseParameters>)paramet
     return clickableViews;
 }
 
-- (BannerSize)adSizeFromAdFormat:(MAAdFormat *)adFormat
+- (VungleAdSize *)adSizeFromAdFormat:(MAAdFormat *)adFormat
+                          parameters:(id<MAAdapterParameters>)parameters
 {
-    if ( adFormat == MAAdFormat.banner )
+    BOOL isAdaptiveBanner = parameters.localExtraParameters[@"adaptive_banner"];
+    if ( isAdaptiveBanner )
     {
-        return BannerSizeRegular;
+        return [VungleAdSize VungleAdSizeWithWidth:[self adaptiveBannerWidthFromParameters: parameters]];
     }
-    else if ( adFormat == MAAdFormat.leader )
-    {
-        return BannerSizeLeaderboard;
-    }
-    else if ( adFormat == MAAdFormat.mrec )
+    else
     {
-        return BannerSizeMrec;
+        if ( adFormat == MAAdFormat.banner )
+        {
+            return [VungleAdSize VungleAdSizeBannerRegular];
+        }
+        else if ( adFormat == MAAdFormat.leader )
+        {
+            return [VungleAdSize VungleAdSizeLeaderboard];
+        }
+        else if ( adFormat == MAAdFormat.mrec )
+        {
+            return [VungleAdSize VungleAdSizeMREC];
+        }
+        else
+        {
+            [NSException raise: NSInvalidArgumentException format: @"Unsupported ad format: %@", adFormat];
+            return [VungleAdSize VungleAdSizeBannerRegular];
+        }
     }
-    else
+}
+
+- (CGFloat)adaptiveBannerWidthFromParameters:(id<MAAdapterParameters>)parameters
+{
+    NSNumber *customWidth = [parameters.localExtraParameters al_numberForKey: @"adaptive_banner_width"];
+    if ( customWidth != nil )
     {
-        [NSException raise: NSInvalidArgumentException format: @"Unsupported ad format: %@", adFormat];
-        return BannerSizeRegular;
+        return customWidth.floatValue;
```

Comment:
Check customWidth value type?

Reply:
`al_numberForKey` should covert it to `NSNumber` if the value exists for the key. Or, it should be `nil`.

Reply:
This API is removed.